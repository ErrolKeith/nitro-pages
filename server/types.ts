export interface BusinessClientGlobals {
  businessId: string; //uuidv4
  sites: string[]; //uuidv4[]
  companyName: string;
  phoneNumber: string;
}

export interface SiteGlobals {
  businessId: string; //uuidv4
  siteId: string;
  companyName: string;
  phoneNumber: string;
  blogSettings: {
    blogTitle: string;
  };
}
