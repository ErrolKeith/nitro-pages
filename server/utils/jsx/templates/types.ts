const templates = ["home", "page", "internal", "404", "index", "post"] as const;
type TemplateType = (typeof templates)[number];

export interface TemplateConfig {
  type: TemplateType;
  siteId: string;
  companyName: string;
  seoTitle: string;
  content: PageContentSection[];
}
