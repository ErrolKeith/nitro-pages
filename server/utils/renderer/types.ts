import { SiteGlobals } from "~/types";
import { TemplateType } from "../jsx/template";

export interface PageContext {
  template: TemplateType;
  slug: string;
  title: string;
  categories: string[];
  content: PageContentSection[];
}

export interface BaseRenderConfig {
  template: TemplateType;
  clientGlobals: SiteGlobals;
  pageContext: PageContext;
}

export interface PageRenderConfig extends BaseRenderConfig {
  template: "page" | "internal" | "post" | "index";
}

export type HomeRenderConfig = Omit<BaseRenderConfig, "template">;
