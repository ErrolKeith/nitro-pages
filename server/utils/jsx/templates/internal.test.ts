import { it, expect } from "vitest";
import { makeInternalTemplate } from "./internal";
import { TemplateConfig } from "./types";

it("can make an internal page", () => {
    const testInternalTemplateConfig: TemplateConfig = {
        type: "internal",
        siteId: "124246796763",
        companyName: "sample company name",
        seoTitle: "sample internal",
        content: [],
      };
      const templateInternalTest = makeInternalTemplate(
        testInternalTemplateConfig
      );
      expect(templateInternalTest).toMatchSnapshot();
});
