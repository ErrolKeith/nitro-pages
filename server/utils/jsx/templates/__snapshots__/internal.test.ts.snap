// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make an internal page 1`] = `
<html
  data-site-id="124246796763"
>
  <head>
    <title>
      sample internal
    </title>
  </head>
  <main />
</html>
`;
